### PR TITLE
Enterprise server (Alpha 24)

### PR DESCRIPTION
**Pull request by chainlink**
TODO: Add option to deploy pachd and enterprise server (or one or the other)
- [ ] ingress
- [ ] pachd-peer?
Depends on: https://github.com/pachyderm/helmchart/pull/122

### CODE REVIEW

**Comment by chainlink** [comment]
@actgardner It looks like the enterprise server is still trying to read the main pachd configuration (getting STORAGE_BACKEND required still)

**Comment by chainlink** [comment]
Fixed in: https://github.com/pachyderm/pachyderm/pull/6393

**Comment by chainlink** [comment]
@actgardner this is the initial config for a separate enterprise server. I've tried it in minikube and it looks like it comes up successfully. There's currently no env vars included, but if there's some we'd like to plumb through that's fine too. Let me know if I'm missing anything (ingress? etc)

**Comment by actgardner** [comment]
The enterprise server uses some env vars. I think the biggest open question is around switching to kube dns in 2.0 (so we could switch ETCD_SERVICE_HOST to `etcd:2379` for instance). Right now we use the following configs:

- `IDENTITY_SERVER_DATABASE`
- `IDENTITY_SERVER_USER`
- `IDENTITY_SERVER_PASSWORD`
- `SESSION_DURATION_MINUTES`
- `PACHYDERM_ENTERPRISE_ETCD_PREFIX` (we might be able to remove the enterprise etcd prefixes altogether)
- `CLUSTER_DEPLOYMENT_ID`
- `ETCD_PREFIX`
- `LOG_LEVEL`
- `POSTGRES_DATABASE_NAME`
- `POSTGRES_SERVICE_SSL`
- `OIDC_PORT`
- `GC_PERCENT`
- `PORT`
- `PEER_PORT`

**Comment by chainlink** [comment]
This is what we have for ingress for the enterprise-server-in-pachd case https://github.com/pachyderm/helmchart/blob/v2.0.x/pachyderm/templates/dash/ingress.yaml#L29-L36

Would we also need this in the standalone enterprise server

(Right now the ingress annotations are tied to traefik for hub, but I'm going to make the ingress more generic)

**Comment by chainlink** [comment]
I'll add in those other Env vars and make em configurable

**Comment by chainlink** [comment]
Manifests now pass tests in main repo. Going to do remaining tasks as follow on.

**Review by actgardner — APPROVED** [review]
LGTM!

File: pachyderm/templates/pachd/imagepull-secret.yaml [hunk 1]
```diff
@@ -2,12 +2,12 @@
...
+{{- if and .Values.pachd.enabled .Values.imageCredentials}}
 apiVersion: v1
 kind: Secret
 metadata:
   name: myregistrykey
 type: kubernetes.io/dockerconfigjson
 data:
   .dockerconfigjson: {{ template "imagePullSecret" . }}
-{{ end }}
+{{- end }}
```
Comment (robert-uhl) [inline pachyderm/templates/pachd/imagepull-secret.yaml:13]: Should this be `{{ end -}}`?

**Review by robert-uhl — APPROVED** [review]

**Comment by chainlink** [comment]
TODO: Change to `deployTarget`

**Pull request merged**